src/arrayMethodReduce.js

```diff
@@ -5,7 +5,22 @@
  */
 function applyCustomReduce() {
   [].__proto__.reduce2 = function(callback, initialValue) {
-    // write code here
+    let prevCallResult;
+    let i = 0;
```

Comment:
there is better use ternary operator
if an initial value exists return 0. If not - return 1

Reply:
@AlinaMatuschak but what if undefined explicitly passed to an initial value?